### ISSUE TITLE
Can methods be passed a molecule or file to molecule?

### ISSUE THREAD

**Issue by shivupa** [chemreps/chemreps#21]
The methods currently work like this:
```
filename = "/path/to/file"
coulomb_matrix(filename)
bag_of_bonds(filename)
```
someone could have already made a molecule class and want to pass that
```
filename = "/path/to/file"
my_mol = Molecule(filename)
coulomb_matrix(my_mol)
bag_of_bonds(my_mol)
```
There's no real reason to support this, but it wouldn't be hard to support it.

**Comment by amandadumi** [comment]
It makes sense. Potentially making it easier for any user to explore different representations more quickly/efficiently. It would just check the data type in the init and go from there, correct?

**Comment by shivupa** [comment]
yeah i think like if type(filename) is str then make a molecule or else try to proceed

**Comment by amandadumi** [comment]
Yeah. There is some function I think called isinstance which can check if the object you’re passing as the first argument is a class instance, given as the second argument.

**Comment by amandadumi** [comment]
must make sure we check for sdf information
